### ISSUE TITLE
Unable to deliver data using reliable/durable QoS [13556]

### ISSUE THREAD

**Issue by oleksandr-hryshchuk** [eProsima/Fast-DDS#2423]
There is an issue when try to deliver data using following DDS DataReader and DataWriter QoS:
- Durability = TRANSIENT_LOCAL
- Reliability = RELIABLE

## Expected Behavior
It should be always possible to deliver data using reliable/durable DataReader and DataWriter even inside a network with some packet loss.

## Current Behavior
It is not always possible to deliver such data samples. The test program hangs after some time (< 100 iterations).

## Steps to Reproduce
Enable the packet loss feature on the loopback interface:
```
sudo tc qdisc add dev lo root netem loss 30%
```
1. Compile the test program (attached).
2. Copy certs directory where you want to run the program.
3. Run the reader instance: `./durability_reliability_test`
4. Run the writer instance: `./durability_reliability_test writer`. 

The writer instance creates the DataWriter with RELIABLE and TRANIENT_LOCAL QoS. Then it writes to the topic a single message.
The reader instance creates the DataReader with RELIABLE and TRANIENT_LOCAL QoS in a cycle and attempts to read the message. Also it re-creates DomainParticipant and Subscriber entities every time.

## System information
- Fast-RTPS version: 2.5.0
- OS: Ubuntu 20.04
- Network interfaces: Loopback

[durability_reliability_test.zip](https://github.com/eProsima/Fast-DDS/files/7914860/durability_reliability_test.zip)

**Comment by oleksandr-hryshchuk** [comment]
There is a recent code change in the function ReaderProxy::requested_changes_set():
```
                    else if (sit > min_seq_in_history)
                    {
                        assert(sit > changes_low_mark_);
                        gap_builder.add(sit);
                    }
```
After I changed it to:
```
                    else if (sit >= min_seq_in_history)
                    {
                        assert(sit > changes_low_mark_);
                        gap_builder.add(sit);
                    }
```
The problem gone and I can pass 100 iterations of the test program with packet loss 30%.
Seems like a regression issue.

**Comment by richiware** [comment]
Thanks for your good report. We could reproduce you problem using your application and find out you're right. The `CacheChange_t` is not added in the GAP message in the case it is irrelevant for a remote reader.

We've created #2448 to fix it in `master` and we will create backports.

Thanks.

**Comment by MiguelCompany** [comment]
This should have been fixed by #2448, which has also been backported on #2465, so closing.
